policy/src/main/java/org/apache/brooklyn/policy/location/CreateLocationPolicy.java

```diff
@@ -198,25 +216,35 @@ public void onEvent(SensorEvent<Boolean> event) {
                 Boolean status = event.getValue();
                 if (status) {
                     if (created.compareAndSet(false, true)) {
-                        LOG.info("Creating new location {} of type {}", getLocationCatalogItemId(), getLocationType().getSimpleName());
-                        LocationSpec spec = LocationSpec.create(getLocationType());
-                        Map<String,AttributeSensor<?>> configuration = getLocationConfiguration();
-                        for (String configKey : configuration.keySet()) {
-                            AttributeSensor<?> sensor = configuration.get(configKey);
-                            Object value = entity.sensors().get(sensor);
-                            spec.configure(configKey, value);
+                        LOG.info("Creating new location {} of type {}", getLocationCatalogItemId(), getLocationType());
+
+                        ImmutableList.Builder<String> builder = ImmutableList.<String>builder().add(
+                                "brooklyn.catalog:",
+                                "  id: " + getLocationCatalogItemId(),
+                                "  itemType: location",
+                                "  item:",
+                                "    type: " + getLocationType(),
+                                "    brooklyn.config:",
+                                "      displayName: " + getDisplayName());
+
+                        if (getLocationConfiguration().size() > 0) {
+                            for (Map.Entry<String, AttributeSensor<?>> entry : getLocationConfiguration().entrySet()) {
+                                AttributeSensor<?> sensor = getLocationConfiguration().get(entry.getKey());
+                                Object value = entity.sensors().get(sensor);
```

Comment:
Interesting, hadn't thought of doing it that way. I assumed that once `service.isUp` was set we would have sensor data, but you're right this may not always be the case.